pkg/gossip/gossip.go

```diff
@@ -840,21 +840,8 @@ func (g *Gossip) updateNodeAddress(key string, content roachpb.Value) {
 		log.Infof(ctx, "removing n%d which was at same address (%s) as new node %v",
 			oldNodeID, desc.Address, desc)
 		g.removeNodeDescriptorLocked(oldNodeID)
```

Comment:
Why isn't this also gone?

Reply:
I figured that that gossip entry is known to be stale at that point so it may as well be removed from gossip until a fresh one is received.